### PR TITLE
allow specifying custom certificates for adapters

### LINKED ISSUE #196: Hono: Use adapter custom certificates in registry service clients.
It's currently not possible to use the `adapters.tenantSpec`, `adapters.deviceRegistrationSpec`, `adapters.credentialsSpec`, `adapters.deviceConnectionSpec` properties of `values.yaml` to configure the registry service clients.
Those are ignore when `deviceRegistryExample.enabled=false`, but then you're not deploying a device registry to begin with.

### PR DESCRIPTION
**Pull request by vhdirk**
Fixes #196

### CODE REVIEW

File: charts/hono/templates/_helpers.tpl [hunk 1]
```diff
@@ -206,7 +206,11 @@ command:
   {{- required ".Values.adapters.commandAndControlSpec MUST be set if example AMQP Messaging Network is disabled" .dot.Values.adapters.commandAndControlSpec | toYaml | nindent 2 }}
 {{- end }}
 tenant:
-{{- if .dot.Values.deviceRegistryExample.enabled }}
+{{- if .dot.Values.adapters.tenantSpec }}
+  {{- range $key, $value := .dot.Values.adapters.tenantSpec }}
+  {{ $key }}: {{ $value }}
+  {{- end }}
```
Comment (sophokles73) [inline charts/hono/templates/_helpers.tpl:212]: how about simply using
```
{{- .dot.Values.adapters.tenantSpec | toYaml | nindent 2 }}
```

Comment (vhdirk) [reply charts/hono/templates/_helpers.tpl]: That's cleaner indeed. Also replaced other occurances.

**Review by sophokles73 — APPROVED** [review]
LGTM

**Comment by sophokles73** [comment]
Thanks for contributing :+1:

**Pull request merged**